main.cc

```diff
@@ -1148,6 +1148,11 @@ To start the scylla server proper, simply invoke as: scylla server (or just scyl
             auth_prep_cache_config.refresh = std::chrono::milliseconds(cfg->permissions_update_interval_in_ms());
 
             qp.start(std::ref(proxy), std::ref(forward_service), std::move(local_data_dict), std::ref(mm_notifier), std::ref(mm), qp_mcfg, std::ref(cql_config), std::move(auth_prep_cache_config), std::ref(group0_client)).get();
+
+            ss.invoke_on_all([&] (service::storage_service& ss) {
+                ss.set_query_processor(qp.local());
+            }).get();
```

Comment:
This is needed to let `storage_service::load_tablet_metadata()` below call `replica::read_tablet_metadata()` which, in turn (https://github.com/scylladb/scylladb/pull/13387/commits/29fd907dee0b686a5aaef45e1718313838a7f267) needs to do `qp.query_internal("select * from system.tablets", ...)`.

This can solved by keeping at least the `qp.query_internal("select * from system.tablets", ...)` part in `system_keyspace::load_tablets()` non-static method. The storage_service already has system_keyspace instance reference, so it will be able to call the proposed load_tablets() w/o the need to plug query processor into itself.

Reply:
If not system keyspace, everything that needs to keep data in system.* tables depends on query_processor (or qctx) and that's not any better.

System keyspace, in turn, only depends on query processor and database, but the latter can be removed.

Reply:
I'm not thinking about run-time dependencies but code dependencies. system_keyspace.hh includes now the raft stuff, topology changes stuff, etc. It would now also include tablets.hh. Anything which deals with system keyspace, and includes system_keyspace.hh, would transitively pull these too. It's becoming a central hub for all the features.

Reply:
Code dependencies can be fixed by splitting system_keyspace.[hc] into multiple files, but preserving runtime dependency graph, no?

Reply:
Isn't what you suggesting equivalent to putting component-specific code in the component-specific .cc/.hh, like in this patch? https://github.com/scylladb/scylladb/pull/13387/commits/9d786c1ebc303373bf739a5d47ec2eeb5669b3b8